### PR TITLE
Update RosMsgs.cs

### PR DESCRIPTION
**Pull request by batouly**
Modify : Marker msg
Add msgs : Duration, StdFloat64, MultiArrayDimension, MultiArrayLayout, StdFloat64MultiArray, PoseArray, CompressedImage, UVCoordinate, MeshFile, RadarReturn, RadarScan

### CODE REVIEW

File: Runtime/RosMsgs.cs [hunk 1]
```diff
@@ -194,6 +258,16 @@ public class Pose : IRosMsg
         public Point position = new Point();
         public Quaternion orientation = new Quaternion() { w = 1 };
     }
+    public class PoseArray : IRosMsg, IStamped
+    {
+        /// <summary>
+        /// An array of poses with a header for global reference.
+        /// </summary>
```
Comment (Sh1co) [inline Runtime/RosMsgs.cs:265]: This comment isn't really necesary or should be placed on top of the class.

File: Runtime/RosMsgs.cs [hunk 2]
```diff
@@ -412,41 +504,159 @@ public class Path : IRosMsg, IStamped
...
 {
+
+    public class UVCoordinate : IRosMsg
+    {
+        string IRosMsg.GetRosType() { return "visualization_msgs.msg.UVCoordinate"; }
+
+        /// <summary>
+        /// Location of the pixel as a ratio of the width of a 2D texture.
+        /// Values should be in range: [0.0-1.0].
+        /// </summary>
```
Comment (Sh1co) [inline Runtime/RosMsgs.cs:515]: Can be a regular comment, because like this, it will show as a tip for variable u only. Or maybe duplicate it for both.

File: Runtime/RosMsgs.cs [hunk 3]
```diff
@@ -412,41 +504,159 @@ public class Path : IRosMsg, IStamped
...
+        public float u;
+        public float v;
+
+    }
+    public class MeshFile : IRosMsg
+    {
+        /// <summary>
+        /// Used to send raw mesh files.
+        /// </summary>
```
Comment (Sh1co) [inline Runtime/RosMsgs.cs:524]: This comment isn't really necesary or should be placed on top of the class.

**Review by Sh1co — APPROVED** [review]
Looks good to me! I only have minor style changes

File: Runtime/RosMsgs.cs [hunk 4]
```diff
@@ -7,6 +7,14 @@ public interface IRosMsg
     {
         public string GetRosType();
     }
+    public class Duration : IRosMsg
+    {
+        string IRosMsg.GetRosType() { return "builtin_interfaces.msg.Duration"; }
```
Comment (xovobobo) [inline Runtime/RosMsgs.cs:12]: Need to handle ROS1 case.
There are will be little bit different fields and message type. 
msg type is **std_msgs.msg.Duration**
and fields: **secs**,**nsecs**

File: Runtime/RosMsgs.cs [hunk 5]
```diff
@@ -412,41 +501,161 @@ public class Path : IRosMsg, IStamped
...
+        /// <summary>
+        /// This stores the raw text of the mesh file.
+        /// </summary>
+        public byte[] data;
+    }
     public class Marker : IRosMsg, IStamped
     {
         string IRosMsg.GetRosType() { return "visualization_msgs.msg.Marker"; }
 
-        public const byte ARROW = 0;
```
Comment (xovobobo) [inline Runtime/RosMsgs.cs:419]: all of that constants should have **byte** type. Why did you change it to **int**?
based on ROS wiki: 
изображение

File: Runtime/RosMsgs.cs [hunk 6]
```diff
@@ -412,41 +501,161 @@ public class Path : IRosMsg, IStamped
...
+        /// Color of the object; in the range: [0.0-1.0]
+        /// </summary>
         public ColorRGBA color;
-        public Time lifetime;
+
+        /// <summary>
+        /// How long the object should last before being automatically deleted.
+        /// 0 indicates forever.
+        /// </summary>
+        public Duration lifetime;
```
Comment (xovobobo) [inline Runtime/RosMsgs.cs:607]: need to handle ROS1 case

File: Runtime/RosMsgs.cs (hunk 5 above)
Comment (xovobobo) [reply Runtime/RosMsgs.cs:419]: > On my device I have this [version ](https://docs.ros.org/en/humble/p/visualization_msgs/interfaces/msg/Marker.html)of the msg

I have found a difference. For ROS2 it should be int32, for ros1 it should be uint8
http://docs.ros.org/en/noetic/api/visualization_msgs/html/msg/Marker.html

Comment (batouly) [reply Runtime/RosMsgs.cs:419]: There are also extra fields in ROS2 msg:
image

File: Runtime/RosMsgs.cs (hunk 4 above)
Comment (batouly) [reply Runtime/RosMsgs.cs:12]: Are the fields secs,nsecs or sec,nsec in ROS1?
image

File: Runtime/RosMsgs.cs [hunk 7]
```diff
@@ -412,44 +540,267 @@ public class Path : IRosMsg, IStamped
...
         public string mesh_resource;
+        public MeshFile mesh_file;
         public bool mesh_use_embedded_materials;
     }
+#else
+    public class Marker : IRosMsg, IStamped
+    {
+        string IRosMsg.GetRosType() { return "visualization_msgs.msg.Marker"; }
+        unit8
+        public const uint ARROW = 0;
```
Comment (xovobobo) [inline Runtime/RosMsgs.cs:706]: Instead of using the **uint** type, the **byte** type should be used.

> uint = uint32
> uint8 = byte

**Review by xovobobo — CHANGES_REQUESTED** [review]

**Review by xovobobo — APPROVED** [review]
